kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/Contributor.java

```diff
@@ -7,46 +7,39 @@
 
 import io.kroxylicious.proxy.config.BaseConfig;
 
+import edu.umd.cs.findbugs.annotations.NonNull;
+
 /**
  * Support loading an Instance of a service, optionally providing it with configuration obtained
  * from the Kroxylicious configuration file.
  *
- * @param <T> the service type
+ * @param <S> the service type
+ * @param <C> the context type
  */
-public interface Contributor<T, S extends Context> {
+public interface Contributor<S, B extends BaseConfig, C extends Context<B>> {
 
     /**
-     * Identifies if this Contributor instance can create instances for the provided short name.
-     * @param typeName The type name to test for.
-     * @return <code>true</code> if the <code>typeName</code> is supported by this contributor.
+     * Identifies the type name this contributor offers.
+     * @return typeName
      */
-    boolean contributes(String typeName);
+    @NonNull
+    String getTypeName();
 
-    /**
-     * @deprecated subsumed into {@link ConfigurationDefinition} and thus {@link Contributor#getConfigDefinition(String)}
-     *
-     * Gets the concrete type of the configuration required by this service instance.
-     * @param typeName service short name
-     *
-     * @return class of a concrete type, or null if this contributor does not offer this short name.
-     */
-    @Deprecated(forRemoval = true, since = "0.3.0")
-    Class<? extends BaseConfig> getConfigType(String typeName);
+    @NonNull
+    Class<B> getConfigType();
```

Comment:
BaseConfig.class seems weird.
* forcing service implementation providers to extend a class seems to have little value.  
* in the case of service implementations that don't require configuration, it seems weird to make them service implementation refer to BaseConfig.

How about eliminating it?
https://github.com/robobario/kproxy/pull/10


I think there should be a default implementation returning BaseConfig.class.  That would relieve filter implementation that don't require config from implementing this method at all.  This would hide

Reply:
Will look at incorporating that PR today